src/main/java/Main.java

```diff
@@ -1,29 +1,35 @@
 import org.apache.commons.codec.binary.Base64InputStream;
 
 import java.io.*;
+import java.util.Properties;
 import java.util.zip.GZIPInputStream;
 
 public class Main {
     public static void main(String[] args) {
-        File file = new File(Main.class.getResource("file.txt.gz").getFile());
-        byte[] buffer = new byte[1024];
-        int signal;
-        String decodedString = "";
         try {
-            InputStream fileInputStream = new FileInputStream(file);
-            GZIPInputStream gzipInputStream = new GZIPInputStream(fileInputStream);
-            Base64InputStream base64InputStream = new Base64InputStream(gzipInputStream);
-            signal = base64InputStream.read(buffer);
-            while (signal != -1) {
-                decodedString += new String(buffer);
-                signal = base64InputStream.read(buffer);
-            }
-            gzipInputStream.close();
-        } catch (FileNotFoundException e) {
-            e.printStackTrace();
+            writeOutput();
         } catch (IOException e) {
             e.printStackTrace();
         }
-        System.out.println(decodedString);
+    }
+
+    private static Properties readProperties() throws IOException {
+        PropertiesReader propertiesReader = new PropertiesReader();
+        return propertiesReader.read("conf.properties");
+    }
+
+    private static InputStreamReader createInputStreamPath(Properties properties) throws IOException {
+        File file = new File(Main.class.getResource(properties.getProperty("inputFile")).getFile());
+        FileInputStream fileInputStream = new FileInputStream(file);
+        GZIPInputStream gzipInputStream = new GZIPInputStream(fileInputStream);
+        Base64InputStream base64InputStream = new Base64InputStream(gzipInputStream);
+        return new InputStreamReader(base64InputStream);
+    }
+
+    private static void writeOutput() throws IOException {
+        Properties properties = readProperties();
+        InputStreamReader inputStreamReader = createInputStreamPath(properties);
+        FileOutputStream fileOutputStream = new FileOutputStream(properties.getProperty("outputFile"));
```

Comment:
why not just use `IOUtil`'s `copy` method on the resulting `InputStream` and `OutputStream`? is there a need to convert the data to an array in the process?